### PR TITLE
feat: signature sanitization

### PR DESCRIPTION
**Pull request by marktoda**
This commit adds basic signature validation, using the ethers signature split/join helpers. This adds support for some signature types such as using the recoveryParam as the v value

ethers ref:
https://github.com/ethers-io/ethers.js/blob/bf0b468490cb293cd916e4fff06e0909273719e6/packages/bytes/src.ts/index.ts#L367

### CODE REVIEW

File: test/uniswapTrades.test.ts [hunk 1]
```diff
@@ -97,6 +97,49 @@ describe('Uniswap', () => {
       expect(methodParameters.value).to.eq(hexToDecimalString(methodParametersV2.value))
     })
 
+    it.only('encodes a single exactInput USDC->ETH swap with EIP-2098 permit', async () => {
```
Comment (zzmp) [inline test/uniswapTrades.test.ts:100]: rm `only`s

File: test/uniswapTrades.test.ts [hunk 2]
```diff
@@ -97,6 +97,49 @@ describe('Uniswap', () => {
...
+      // append recoveryParam as v
+      signature += BigNumber.from(recoveryParam).toHexString().slice(2)
+      // assert ethers sanitization technique works
+      expect(utils.joinSignature(utils.splitSignature(signature))).to.eq(originalSignature)
+      const opts = swapOptions({ inputTokenPermit: toInputPermit(signature, permit) })
+      const methodParameters = SwapRouter.swapERC20CallParameters(buildTrade([trade]), opts)
+      const methodParametersV2 = SwapRouter.swapCallParameters(new UniswapTrade(buildTrade([trade]), opts))
+      registerFixture('_UNISWAP_V2_1000_USDC_FOR_ETH_PERMIT_V_RECOVERY_PARAM', methodParameters)
+      expect(hexToDecimalString(methodParameters.value)).to.eq('0')
+      expect(methodParameters.calldata).to.eq(methodParametersV2.calldata)
```
Comment (azflin) [inline test/uniswapTrades.test.ts:139]: Why do you do this? Why do we need to make sure `swapERC20CallParameters()` (which the docs say is deprecated) and `swapCallParameters()` produce the same result

**Review by azflin — APPROVED** [review]
LGTM. Just have that one commented question.

File: test/uniswapTrades.test.ts (hunk 2 above)
Comment (marktoda) [reply test/uniswapTrades.test.ts:139]: I think because the deprecated API is still used, so we want to ensure we keep compatibility

**Pull request merged**